lib/model/query/entities.js

```diff
@@ -65,10 +77,16 @@ const nextval = sql`nextval(pg_get_serial_sequence('entities', 'id'))`;
 // standard authenticated API request. The worker has better access to the event
 // actor/initiator and actee/target so it will do the logging itself (including
 // error logging).
-const createNew = (dataset, partial, subDef, sourceId, userAgentIn) => ({ one, context }) => {
+const createNew = (dataset, partial, subDef, sourceId, userAgentIn) => async ({ one, context, maybeOneFirst }) => {
   let creatorId;
   let userAgent;
 
+  // Validate UUID against purged entities
+  (await _isPurgedUuid(maybeOneFirst, partial.uuid))
+    .ifDefined(() => {
+      throw Problem.user.uniquenessViolation({ fields: 'uuid', values: partial.uuid });
```

Comment:
I think this should throw this instead
```suggestion
      throw Problem.user.entityDeleted({ entityUuid: clientEntity.uuid });
```

But I also noticed that if you try to reuse a soft-deleted UUID, it also has `Problem.user.uniquenessViolation` instead of `Problem.user.entityDeleted`.

Reply:
I have introduced `entityUniquenessViolationWithDeleted` - please suggest a shorter name. 

I think we should not return 404 during create entity where there is/was entity with the same UUID is soft-deleted or purged. So for create I am returning 409 and for update 404.

Let me know what do you think about this.

Reply:
I see what you mean about the 409 vs the 404. Also that other `entityDeleted` problem doesn't have a very descriptive name; it's more like `entityNotFoundDeleted`. It's also only used in one case where a submission tries to update something about an entity and the entity has been deleted. 

I can't really think of a shorter name than `entityUniquenessViolationWithDeleted`. `entityConflictDeleted`? Nah, what you have, even though it is long, is clear.